R/get_hospdata.R

```diff
@@ -0,0 +1,159 @@
+    to = as.Date(max(ecdc_data$date, owid_data$date)),
+    by = "day"
+  )
+
+  # complete by date grouped within country/indicator/source
+  hospdata_long <-
+    hospdata %>%
+    mutate(date = lubridate::ymd(date)) %>%
+    group_by(iso_code, indicator, source) %>% # for each indicator & source
+    tidyr::complete(date = all_dates) %>% # complete by country and date
+    mutate(carry_fwd_value = zoo::na.locf(value, na.rm = F, maxgap = 14)) %>%
+    arrange(iso_code, date) %>%
+    ungroup() %>%
+    rename(id = iso_code)
+
+
+  return(hospdata_long)
+}
+
+#' Get hospital data in a wide format where indicators are columns
+#'
+#' @param hospdata_long the hospital data in long format as generated
+#'  by \code{\link{get_hospdata_long}}.
+#' @param preferred_source character or character vector, either "OWID",
+#'  "ECDC", or c("OWID", "ECDC") for both). Defaults to keeping both sources.
+#'
+#' @return
+#' A data frame with n rows and up to 13 columns:
+#'
+#' \describe{
+#'       \item{id}{character, ISO 3166-1 alpha-3 country code}
+#'       \item{source}{character, ECDC or OWID, see details for more information.}
+#'       \item{date}{date of observation}
+#'       \item{daily_icu_occupancy}{OWID & ECDC: Number of COVID-19 patients in ICU on a given day}
+#'       \item{daily_icu_occupancy_per_million}{OWID: daily_icu_occupancy per million people}
+#'       \item{daily_hospital_occupancy}{OWID & ECDC: Number of COVID-19 patients in hospital on a given day}
+#'       \item{daily_hospital_occupancy_per_million}{OWID: daily_hospital_occupancy per million people}
+#'       \item{weekly_new_hospital_admissions}{OWID & ECDC: Number of COVID-19 patients newly admitted
+#'             to hospitals in a given week}
+#'       \item{weekly_new_hospital_admissions_per_100k}{ECDC: weekly_new_hospital_admissions from ECDC per 100,000 persons}
+#'       \item{weekly_new_hospital_admissions_per_million}{OWID: weekly_new_hospital_admissions from OWID per million people}
+#'       \item{weekly_new_icu_admissions}{OWID & ECDC: Number of COVID-19 patients newly admitted to ICU in a given week}
+#'       \item{weekly_new_icu_admissions_per_million}{OWID: Weekly_new_icu_admissions from OWID per million people}
+#'       \item{weekly_new_icu_admissions_per_100k}{ECDC: weekly_new_icu_admissions from ECDC per 100,000 persons}
+#' }
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'  hospdata_long <- get_hospdata_long()
+#'  hospdata_wide <- get_hospdata_wide(hospdata_long, preferred_source = "OWID")
+#' }
+#'
+get_hospdata_wide <- function(hospdata_long,
+                              preferred_source = c("OWID", "ECDC")) {
+  preferred_source <- match.arg(preferred_source, several.ok = TRUE)
+
+  hospdata_wide <-
+    hospdata_long %>%
+    filter(source %in% preferred_source) %>%
+    select(-carry_fwd_value) %>%
+    tidyr::pivot_wider(names_from = indicator, values_from = value) %>%
+    janitor::clean_names()
```

Comment:
Hm. I might suggest explicitly renaming the columns instead of using `clean_names`, just so we don't have to import an entire package just for that function. I do love janitor, though!

Reply:
Easy approach: write a lookup in lookups.R that maps raw to clean column names, and use dplyr::recode() in a mutate block above the pivot_wider(). 

I did something similar here: https://github.com/CDCgov/SaviR/blob/640c554e049161fefe8b7f5be814048bb8a604e7/R/get_covid_sources.R#L29

We could also write a test in the data_accessors to check that the column names are present as we expect, so we can see if that structure changes.

Reply:
Easy approach: write a lookup in lookups.R that maps raw to clean column names, and use dplyr::recode() in a mutate block above the pivot_wider(). 

I did something similar here: https://github.com/CDCgov/SaviR/blob/640c554e049161fefe8b7f5be814048bb8a604e7/R/get_covid_sources.R#L29

We could also write a test in the data_accessors to check that the column names are present as we expect, so we can see if that structure changes.

Reply:
Oh cool, yeah can try that! You think it's worth keeping 'get_hospdata_wide'? I don't know if the wide format ever gets used down stream at the moment...I think the DUB would just use get_hospdata_long -> get_hospdata_latest. Also I don't quite know why the github actions checks are failing (something to do with a vignette but I didn't think I changed anything in there?)!

Reply:
See below RE: wide and general API stuff.

Not sure on the testing! I'm looking into it. May be upstream/unrelated.

Reply:
@mrajeev08 FYI, see #62. Looks like a {gt} error.